### ISSUE TITLE
SharpYuv*Bit/RGBToYUVTest.* in avifrgbtoyuvtest fail if libsharpyuv is not found

### ISSUE THREAD

**Issue by wantehchang** [AOMediaCodec/libavif#1164]
Steps to reproduce:

1. Do not install libsharpyuv on the system.
2. In the `ext/` directory, run `./aom.cmd` and `./googletest.cmd`.
3. Create a `build/` directory and change into that directory.
4. Run `cmake -G Ninja .. -DBUILD_SHARED_LIBS=OFF -DAVIF_CODEC_AOM=ON -DAVIF_LOCAL_AOM=ON -DAVIF_BUILD_TESTS=ON -DAVIF_LOCAL_GTEST=ON`
5. Run `ninja`
6. Run `tests/avifrgbtoyuvtest`

The `SharpYuv*Bit/RGBToYUVTest.*` tests fail. Here are two examples:

```
[----------] 144 tests from SharpYuv8Bit/RGBToYUVTest
[ RUN      ] SharpYuv8Bit/RGBToYUVTest.ConvertWholeRange/0
/usr/local/google/home/wtc/libavif.1/libavif/tests/gtest/avifrgbtoyuvtest.cc:206: Failure
Expected equality of these values:
  avifImageRGBToYUV(yuv.get(), &src_rgb)
    Which is: 25
  AVIF_RESULT_OK
    Which is: 0
[  FAILED  ] SharpYuv8Bit/RGBToYUVTest.ConvertWholeRange/0, where GetParam() = (8, 8, 0, 3, 0, 6, 4, true, 17, 1.2, 34) (0 ms)
[ RUN      ] SharpYuv8Bit/RGBToYUVTest.ConvertWholeRange/1
/usr/local/google/home/wtc/libavif.1/libavif/tests/gtest/avifrgbtoyuvtest.cc:206: Failure
Expected equality of these values:
  avifImageRGBToYUV(yuv.get(), &src_rgb)
    Which is: 25
  AVIF_RESULT_OK
    Which is: 0
[  FAILED  ] SharpYuv8Bit/RGBToYUVTest.ConvertWholeRange/1, where GetParam() = (8, 8, 0, 3, 0, 1, 4, true, 17, 1.2, 34) (0 ms)
```

The reason is that libsharpyuv is not found, so we return AVIF_RESULT_NOT_IMPLEMENTED (25) from the following code in `avifImageRGBToYUV()`:

```
    avifBool converted = AVIF_FALSE;

    // Try converting with libsharpyuv.
    if ((rgb->chromaDownsampling == AVIF_CHROMA_DOWNSAMPLING_SHARP_YUV) && (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420)) {
        const avifResult libSharpYUVResult = avifImageRGBToYUVLibSharpYUV(image, rgb, &state);
        if (libSharpYUVResult != AVIF_RESULT_OK) {
            // Return the error if sharpyuv was requested but failed for any reason, including libsharpyuv not being available.
            return libSharpYUVResult;
        }
        converted = AVIF_TRUE;
    }
```

**Comment by wantehchang** [comment]
@maryla-uc @y-guyon Could you take a look at this bug? I don't have a good solution yet.

@jbeich Did you run into this bug when you tested the v0.11.0 release? You can avoid this bug by installing libsharpyuv on the computer.

**Comment by jbeich** [comment]
> @jbeich Did you run into this bug when you tested libavif v0.11.0?

Yes but libsharpyuv isn't packaged on FreeBSD yet (see also [Repology](https://repology.org/projects/?search=sharpyuv)). I've [proposed libsharpyuv via libwebp package](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=267062) but it requires a big patch until the next release. Another option is bundling ([discouraged](https://docs.freebsd.org/en/books/porters-handbook/special/#bundled-libs) but acceptable) which I haven't tried yet.

**Comment by jzern** [comment]
I'd prefer not to have libsharpyuv propagate until it's part of a formal libwebp release. There will be some changes to that code before 1.3.0, so disabling it might be the safest option.

**Comment by wantehchang** [comment]
I came up with a fix that seems reasonable: https://github.com/AOMediaCodec/libavif/pull/1168.

**Comment by wantehchang** [comment]
@jbeich Could you test https://github.com/AOMediaCodec/libavif/pull/1171 and see if it fixes this bug for you? Thanks.

I have verified https://github.com/AOMediaCodec/libavif/pull/1171 locally.

**Comment by jbeich** [comment]
I confirm, as of 7e70644c4c36 (after #1171 merged) all tests pass on FreeBSD 13.1 amd64 & 12.3 i386.

**Comment by jbeich** [comment]
Oops, my testing was limited to libavif with just libaom and dav1d enabled. When disabling libaom and enabling svt-av1 or rav1e more fail (need to create a new issue).

**Comment by wantehchang** [comment]
Jan: Thanks a lot for the extra testing. I will wait for your new bug report.
